server/resp/src/main/java/org/infinispan/server/resp/commands/connection/AUTH.java

```diff
@@ -25,7 +26,10 @@ public CompletionStage<RespRequestHandler> perform(Resp3AuthHandler handler,
                                                       List<byte[]> arguments) {
       CompletionStage<Boolean> successStage = handler.performAuth(ctx, arguments.get(0), arguments.get(1));
 
-      return handler.stageToReturn(successStage, ctx, success -> createAfterAuthentication(success, handler));
+      return handler.stageToReturn(successStage, ctx, success -> {
+         if (success) Consumers.OK_BICONSUMER.accept(null, handler.allocator());
```

Comment:
I am not a fan of having this here and then the non success writing to the channel in handleAuthResponse. Is there a reason this was moved? We should strive to have the error and success paths when writing a response near each other.

Reply:
The mix with AUTH during the HELLO. In that case, we don't want the `+OK`, but we want the unauthorized response. Let me move some stuff around.